pallets/ocex/src/lib.rs

```diff
@@ -211,11 +215,33 @@ pub mod pallet {
 					));
 				});
 				<Accounts<T>>::insert(&main_account, account_info);
-				Self::deposit_event(Event::MainAccountRegistered { main: main_account, proxy });
+				Self::deposit_event(Event::NewProxyAdded { main: main_account, proxy });
 			}
 			Ok(())
 		}
 
+		/// Removes a proxy account from pre-registered main acocunt
+		#[pallet::weight(10000)]
+		pub fn remove_proxy_account(origin: OriginFor<T>, proxy: T::AccountId) -> DispatchResult {
+			let main_account = ensure_signed(origin)?;
+			ensure!(<Accounts<T>>::contains_key(&main_account), Error::<T>::MainAccountNotFound);
+			<Accounts<T>>::try_mutate(&main_account, |account_info| {
+				if let Some(account_info) = account_info {
+					let proxy_positon = account_info.proxies.iter().position(|account| *account == proxy).ok_or(Error::<T>::ProxyNotFound)?;
+					ensure!(account_info.proxies.len() > 1, Error::<T>::MinimumOneProxyRequired);
```

Comment:
This assertion should be moved to line 230 (before iterating on it).

Reply:
done